wallet-new/test/unit/Test/Spec/Addresses.hs

```diff
@@ -324,6 +324,26 @@ spec = describe "Addresses" $ do
                                     -> pure ()
                            _ -> fail ("Got " ++ show res)
 
+            prop "arbitrary number of addresses, pages and per page" $ withMaxSuccess 500 $ do
+                monadicIO $ do
+                    (rNumOfAddresses :: Int) <- pick $ elements [0..20]
+                    (rNumOfPages :: Int) <- pick $ elements [0..20]
+                    (rNumPerPage :: Int) <- pick $ elements [0..20]
+                    withAddressFixtures rNumOfAddresses $ \_ layer _ correctAddresses -> do
+                        let (!>) = drop . (subtract 1)
+                        let (<!) = flip take
+                        let slice numOfPage numPerPage xs
+                                | numOfPage*numPerPage == 0 = []
+                                | otherwise = ((numOfPage-1)*numPerPage+1) !> xs <! numPerPage
+                        let pp = PaginationParams (Page rNumOfPages) (PerPage rNumPerPage)
+                        res <- runExceptT $ runHandler' $ do
+                           Handlers.listAddresses layer (RequestParams pp)
+                        case res of
+                           Right wr | toCheckAddresses <- wrData wr
+                                    , toCheckAddresses == (map addressFixtureAddress $ slice rNumOfPages rNumPerPage correctAddresses)
+                                    -> pure ()
+                           _ -> fail ("Got " ++ show res)
```

Comment:
This will give us very little information in case of failure. Perhaps we can reduce a bit the ranges of elements above, and simply go for the `===` operator from quickcheck (which prints both sides of the equal sign when it fails).

What do you think?

Reply:
I can change to:
```Right wr | toCheckAddresses <- wrData wr                                                                                                                       
               -> quickCheck $ toCheckAddresses === (map addressFixtureAddress $ slice rNumOfPages rNumPerPage correctAddresses)``` and reduce elements to 15.

Reply:
`quickcheck` is used to run properties, here we are inside a property (even though it's monadic).
Something like `pure (toCheckAddresses === (map addressFixtureAddress $ ...)` should do, right?

Reply:
right!

Reply:
Well, the `case res of` doesn't make much sense here right? 

```hs
let gotAddresses  = wrData <$> res
let wantAddresses = pure $ map addressFixtureAddress $ slice rNumOfPages rNumPerPage fixture)
pure (gotAddresses === wantAddresses)
```